File: Swift/ExampleTests/common/ResultString.swift
public struct ResultString: CustomStringConvertible, Equatable {
    public let addressMatches: String

    public init(addressMatches: String) {
        self.addressMatches = addressMatches
    }

    public init(fromArray v: [String]) {
        self.addressMatches = v.count > 0 ? v[0] : ""
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    public static func fromText(_ text: String) -> ResultString {
        let parts = Tokens.require(text, 1, "Result")
        return ResultString(addressMatches: parts[0])
    }

    public var description: String {
        return Tokens.token(addressMatches)
    }

    public static let dncString = "?DNC?"

    public static func == (a: ResultString, b: ResultString) -> Bool {
        return (a.addressMatches == dncString || b.addressMatches == dncString || a.addressMatches == b.addressMatches)
    }
}

File: JavaScript/ExampleTests/common/ResultString.js
import * as tokens from "./tokens.js";

export class ResultString {
  static DNC_STRING = "?DNC?";

  constructor(addressMatches = "") {
    this.addressMatches = addressMatches;
  }

  static fromList(values) {
    const v = Array.from(values);
    return new ResultString(
      v[0] ?? ""
    );
  }

  /** Builds from the text form, e.g. Money as "25 USD". */
  static fromText(text) {
    const parts = tokens.require_(text, 1, "Result");
    return new ResultString(
      parts[0]
    );
  }

  toString() {
    return tokens.token(this.addressMatches);
  }

  equals(other) {
    if (!(other instanceof ResultString)) return false;
    return (this.addressMatches === ResultString.DNC_STRING || other.addressMatches === ResultString.DNC_STRING || this.addressMatches === other.addressMatches);
  }

  static equalLists(a, b) {
    if (a.length !== b.length) return false;
    return a.every((row, i) => row.equals(b[i]));
  }
}
